src/linker/Linker.Dataflow/XmlFlowAnnotationSource.cs

```diff
@@ -0,0 +1,690 @@
+			if (!String.IsNullOrEmpty (value))
+				ProcessFieldName (type, value, iterator);
+		}
+
+		void ProcessFieldSignature (TypeDefinition type, string signature, XPathNodeIterator iterator)
+		{
+			FieldDefinition field = GetField (type, signature);
+			if (field != null) {
+				ArrayBuilder<Attribute> attributes = ProcessAttributes (iterator.Current.SelectChildren ("attribute", _ns));
+				DynamicallyAccessedMemberTypes fieldAnnotation = ParseKinds (attributes);
+				if (fieldAnnotation != DynamicallyAccessedMemberTypes.None)
+					_fields[field] = fieldAnnotation;
+			}
+		}
+
+		void ProcessFieldName (TypeDefinition type, string name, XPathNodeIterator iterator)
+		{
+			if (!type.HasFields)
+				return;
+
+			foreach (FieldDefinition field in type.Fields) {
+				if (field.Name == name) {
+					ArrayBuilder<Attribute> attributes = ProcessAttributes (iterator.Current.SelectChildren ("attribute", _ns));
+					DynamicallyAccessedMemberTypes fieldAnnotation = ParseKinds (attributes);
+					if (fieldAnnotation != DynamicallyAccessedMemberTypes.None)
+						_fields[field] = fieldAnnotation;
+				}
+			}
+		}
+
+		protected static FieldDefinition GetField (TypeDefinition type, string signature)
+		{
+			if (!type.HasFields)
+				return null;
+
+			foreach (FieldDefinition field in type.Fields)
+				if (signature == GetFieldSignature (field))
+					return field;
+
+			return null;
+		}
+
+		static string GetFieldSignature (FieldDefinition field)
+		{
+			return field.FieldType.FullName + " " + field.Name;
+		}
+
+		void ProcessMethods (TypeDefinition type, XPathNodeIterator iterator)
+		{
+			while (iterator.MoveNext ())
+				ProcessMethod (type, iterator);
+		}
+
+		protected virtual void ProcessMethod (TypeDefinition type, XPathNodeIterator iterator)
+		{
+#if !FEATURE_ILLINK
+			if (IsExcluded (iterator.Current))
+				return;
+#endif
+
+			string value = GetSignature (iterator.Current);
+			if (!String.IsNullOrEmpty (value))
+				ProcessMethodSignature (type, value, iterator);
+
+			value = GetAttribute (iterator.Current, "name");
+			if (!String.IsNullOrEmpty (value))
+				ProcessMethodName (type, value, iterator);
+		}
+
+		void ProcessMethodSignature (TypeDefinition type, string signature, XPathNodeIterator iterator)
+		{
+			MethodDefinition meth = GetMethod (type, signature);
+			if (meth != null)
+				ProcessMethodChildren (type, meth, iterator);
+		}
+
+		void ProcessMethodChildren (TypeDefinition type, MethodDefinition method, XPathNodeIterator iterator)
+		{
+			ArrayBuilder<Attribute> attributes = ProcessAttributes (iterator.Current.SelectChildren ("attribute", _ns));
+			ArrayBuilder<(string, ArrayBuilder<Attribute>)> parameterAnnotations = ProcessParameters (type,
+				method, iterator.Current.SelectChildren ("parameter", _ns));
+			ArrayBuilder<ArrayBuilder<Attribute>> returnParameterAnnotations = ProcessReturnParameters (type,
+				method, iterator.Current.SelectChildren ("returnparameter", _ns));
```

Comment:
I would prefer either `return` or `returnParameter` names (personally I like `return` better, but either is fine)